stratum/hal/lib/tdi/es2k/es2k_sde_packetio.cc

```diff
@@ -0,0 +1,428 @@
+  uint64_t nb_pkts;
+  std::vector<uint64_t> pkt_len;  /* len of pkts */
+  std::vector<uint64_t> pkt_data; /* pointers to pkts data */
+
+  // parse callback params
+  pkts_info_recv->getValue(PORT_ID, &port_id);
+  pkts_info_recv->getValue(QUEUE_ID, &queue_id);
+  pkts_info_recv->getValue(NB_PKTS, &nb_pkts);
+  pkts_info_recv->getValue(PKT_LEN, &pkt_len);
+  pkts_info_recv->getValue(PKT_DATA, &pkt_data);
+
+  printf("[satish] pktio: %lu pkts received on port_id: %lu, queue_id: %lu\n",
+         nb_pkts, port_id, queue_id);
+
+  Es2kSdeWrapper* tdi_sde_wrapper = Es2kSdeWrapper::GetSingleton();
+
+  const char* raw_data;
+  for (uint64_t i = 0; i < nb_pkts; i++) {
+    printf("pktio: pkt_len: %lu\n", pkt_len[i]);
+    raw_data = reinterpret_cast<const char*>(pkt_data[i]);
+
+    // satish: Only for testing: dump the pkt headers
+    printf("[satish] pktio: pkt received:\n");
+    dump_pkt(raw_data);
+
+    // satish: TODO: Need to find a way to get device_id
+    tdi_sde_wrapper->HandlePacketRx(0, raw_data, pkt_len[i]);
+  }
+}
+
+// StartPacketIo registers callbacks for all the ports and queues in specified
+// in PacketIoConfig
+::util::Status Es2kSdeWrapper::StartPacketIo(int dev_id) {
+  if (pktio_config_.ports_size() == 0) {
+    LOG(INFO) << "pktio not configured";
+    std::cout << "[satish] pktio not configured" << std::endl;
+    return ::util::OkStatus();
+  }
+
+  std::cout << "[satish] pktio configured" << std::endl;
+
+  // create Target
+  const ::tdi::Device* device = nullptr;
+  ::tdi::DevMgr::getInstance().deviceGet(dev_id, &device);
+  std::unique_ptr<::tdi::Target> dev_tgt;
+  device->createTarget(&dev_tgt);
+
+  // get table
+  const ::tdi::Table* table;
+  RETURN_IF_TDI_ERROR(tdi_info_->tableFromNameGet(PKTIO_TABLE_NAME, &table));
+
+  // Allocate rx params
+  std::unique_ptr<::tdi::NotificationParams> rx_notification_params;
+  tdi_status_t status = table->notificationRegistrationParamsAllocate(
+      RX, &rx_notification_params);
+  if (status != BF_SUCCESS) {
+    return MAKE_ERROR(::util::error::Code::INTERNAL)
+           << "rx_notification allocation fail ";
+  }
+
+  // Allocate tx params
+  std::unique_ptr<::tdi::NotificationParams> tx_notification_params;
+  table->notificationRegistrationParamsAllocate(TX, &tx_notification_params);
+  if (status != BF_SUCCESS) {
+    return MAKE_ERROR(::util::error::Code::INTERNAL)
+           << "tx_notification allocation fail ";
+  }
+
+  // Register rx and tx callbacks for all the ports and queues
+  int port_id;
+  for (int port_idx = 0; port_idx < pktio_config_.ports_size(); port_idx++) {
+    port_id = pktio_config_.ports(port_idx);
+    for (int queue_id = 0; queue_id < pktio_config_.nb_rxqs(); queue_id++) {
+      // set values
+      rx_notification_params->setValue(PORT_ID, port_id);
+      rx_notification_params->setValue(QUEUE_ID, queue_id);
+      rx_notification_params->setValue(RX_BURST_FIELD_ID,
+                                       1);  // burst_sz set to 1
```

Comment:
The comment doesn't add value. It's fairly obvious that you're setting the burst size to 1. If there were a comment at all, it would be to say _why_ the burst size is 1, and I don't see that as a pressing question.

Reply:
Done.